src/main/java/com/redhat/pantheon/servlet/ModuleRenderServlet.java

```diff
@@ -0,0 +1,128 @@
+ * Example: <server_url>/api/module/render?locale=en-us&module_id=xyz
+ * The said url is accessible outside of the system without any authentication.
+ *
+ * @author Ankit Gadgil
+ */
+@Component(
+        service = Servlet.class,
+        property = {
+                "sling.servlet.resourceTypes=sling/servlet/default",
+                "sling.servlet.selectors=query",
+                "sling.servlet.extensions=json",
+                Constants.SERVICE_DESCRIPTION + "=Servlet to facilitate GET operation which accepts locale and module uuid to output module data",
+                Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team"
+        })
+@SlingServletPaths(value = "/api/module/render")
+public class ModuleRenderServlet extends AbstractJsonSingleQueryServlet {
+    private final Logger log = LoggerFactory.getLogger(ModuleRenderServlet.class);
+
+    @Override
+    protected String getQuery(SlingHttpServletRequest request) {
+        // Get the query parameter(s)
+        String localeParam = ServletUtils.paramValue(request, "locale", DEFAULT_MODULE_LOCALE.toString());
+        String uuidParam = paramValue(request, "module_id", "");
+        String nodeType = paramValue(request, "nodeType","pant:module");
+
+        StringBuilder query = new StringBuilder("select * from ")
+                .append("[").append(nodeType).append("]")
+                .append(" as module WHERE ")
+//                .append("where (isdescendantnode(module, '/content/repositories') ")
+//                .append("or isdescendantnode(module, '/content/modules') ")
+//                .append("or isdescendantnode(module, '/content/sandbox')) ")
+                // look for a specific uuid for module
+                .append("module.[jcr:uuid] = '" + uuidParam + "'");
+
+        return query.toString();
+    }
+
+    @Override
+    protected Map<String, Object> resourceToMap(@NotNull Resource resource) {
+        Module module = resource.adaptTo(Module.class);
+        Optional<Metadata> draftMetadata = module.getDraftMetadata(DEFAULT_MODULE_LOCALE);
+        Optional<Metadata> releasedMetadata = module.getReleasedMetadata(DEFAULT_MODULE_LOCALE);
+        Optional<Content> draftContent = module.getDraftContent(DEFAULT_MODULE_LOCALE);
+        Optional<Content> releasedContent = module.getReleasedContent(DEFAULT_MODULE_LOCALE);
+        Optional<ModuleRevision> draftRevision = module.getDraftRevision(DEFAULT_MODULE_LOCALE);
+        Optional<ModuleRevision> releasedRevision = module.getReleasedRevision(DEFAULT_MODULE_LOCALE);
+
+        Map<String, Object> moduleMap = super.resourceToMap(resource);
+        Map<String, Object> moduleDetails = new HashMap<>();
+
+        moduleDetails.put("status", "SUCCESS");
+        moduleDetails.put("message", "Module Found");
+
+        String resourcePath = resource.getPath();
+        moduleMap.put("locale", module.getModuleLocale(DEFAULT_MODULE_LOCALE).getName());
```

Comment:
I mentioned this already, but I'll do it again in the review. I think a better idea here, instead of using a `toMap` function, is to simply have a function like
```java
protected Object transform(Resource resource) {
    return resource.toValueMap();
}
```
This way the default transformation is just to a value map, but overrides may transform into anything that can be turned into json.

In our specific case, we create a `Module` DTO of sorts which has all these fields baked in as hava properties. e.g.
```java
// when converted to json this will generate a field of name "title"
String getTitle() {
   return title
}
```
Anyway, this can be a later improvement since our query method already has a `resourceToMap` method.

Reply:
That's a good improvement to look into.